app/src/main/java/org/covidwatch/android/presentation/SetupBluetoothFragment.kt

```diff
@@ -55,6 +59,11 @@ class SetupBluetoothFragment : Fragment(R.layout.fragment_setup_bluetooth),
     }
 
     override fun onPermissionsGranted(requestCode: Int, perms: MutableList<String>) {
-        findNavController().navigate(R.id.homeFragment)
+        permissionGranted()
+    }
+
+    private fun permissionGranted() {
+        userFlowRepository.updateSetupUserFlow()
+        findNavController().popBackStack(R.id.homeFragment, false)
```

Comment:
Does this make it so we won't come back here with the back button?

Reply:
yep :)